lib/services/elasticsearch.js

```diff
@@ -260,9 +261,8 @@ class ElasticSearch extends Service {
     return this.client.get(esRequest)
       .then(result => {
         if (result._source) {
-          if (result._source._kuzzle_info && !result._source._kuzzle_info.active) {
-            // todo Feedback how to get it from the 'trash' once it is implemented
-            return Bluebird.reject(new NotFoundError(`Document ${result._id} was already deleted`));
+          if (result._source._kuzzle_info && !result._source._kuzzle_info.active && !request.input.args.includeTrash) {
+            return Bluebird.reject(new NotFoundError(`Document ${result._id} was already deleted. If you want to also get trashed documents set includeTrash argument to true`));
```

Comment:
I'm not at ease with this error message on a `get` route. I see no reason to mention that the document has been "already deleted", or that it can be retrieved in the trash. At least not with the current error message.
To me, there is a conflict with returning a 404 while indicating that the document somehow exists.

I see several possibilities we need to discuss:
* throw a simple 404 "document not found" error
* throw a new 423 error (resource locked), with a message like: "Cannot get a deleted document without the "includeTrash" flag'. This new error could be, for instance, a ResourceLockedError